[Response Ops][Rules] Add New Rule Form to Stack Management

JiaweiWu opened the pull request:
## Summary

Enables and adds the new rule form to stack management. We are only going to turn this on for stack management for now until we are confident that this is fairly bug free.

### To test:

1. Switch `USE_NEW_RULE_FORM_FEATURE_FLAG` to true
2. Navigate to stack management -> rules list
3. Click "Create rule" 
4. Assert the user is navigated to the new form
5. Create rule
6. Assert the user is navigated to the rule details page
7. Click "Edit"
8. Edit rule
9. Assert the user is navigated to the rule details page
10. Try editing a rule in the rules list and assert everything works as expected

We should also make sure this rule form is not enabled in other solutions.

### Checklist
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios


<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

elasticmachine commented:
Pinging @elastic/response-ops (Team:ResponseOps)

maryam-saeidi commented on packages/kbn-alerts-ui-shared/src/common/constants/rule_form_flag.ts:
```diff
@@ -0,0 +1,10 @@
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+export const USE_NEW_RULE_FORM_FEATURE_FLAG = false;
```
Shouldn't we define this feature flag in Kibana config?

cnasikas replied:
Hey Maryam. It is not something we want users to control. We want to release the new rule form. For that reason, maybe we should remove it entirely. @JiaweiWu Do I miss something?

JiaweiWu replied:
Yea it's just a easy way to not have to fix a bunch of tests and allow folks to test this

maryam-saeidi reviewed (commented):
I changed the value of USE_NEW_RULE_FORM_FEATURE_FLAG to true, but when I tried to create a rule, I got Not Found (URL: http://localhost:5602/app/management/insightsAndAlerting/triggersActions/rules/create/observability.rules.custom_threshold)
Do I need to do something else?

**Update**

I think the issue is related to having a basePath in my config: (The above URL does not have `/kibana` at start)
```
server.basePath: '/kibana'
server.publicBaseUrl: "http://localhost:5601/kibana"
```

cnasikas commented:
@elasticmachine merge upstream

js-jankisalvi commented:
Found an issue for alert delay, I think it's an issue for all rule types. (tested it in log threshold and anomaly detection job health rule types).
UI allows to remove value from alert delay but when you save the rule, previous value is saved.

https://github.com/user-attachments/assets/5d86bdf9-a820-4ff6-9ffa-e6feb9069b3a

cnasikas reviewed (commented):
Hey @JiaweiWu! I tested the Index threshold, Anomaly detection, and Metric threshold rule types and the ServiceNow ITSM, ServiceNow SecOps, and ServiceNow ITOM connector types. I tested various settings in the rule form and in dark mode. I found the following:

- Same bug as here but with the connectors modal: https://github.com/elastic/kibana/issues/184048
- When editing an Anomaly detection rule type I get the following error

![Screenshot 2024-10-07 at 4 32 11 PM](https://github.com/user-attachments/assets/553c446f-4d34-49c3-a42b-0c6b86a8c0c4)

- The form shows errors even if I did not touch any field. We can leave it as enhancement for the future as it probably has to do with how each rule type component initiates their state.

guskovaue commented:
Hi, @JiaweiWu !
During testing ES query rule, I found small issue. Probably it's like this for all rule types.
Look video please:

https://github.com/user-attachments/assets/e45f8ffb-b527-4a06-b3ff-2a3e76f2eaca

cnasikas commented on packages/kbn-alerts-ui-shared/src/common/hooks/use_load_connectors.ts:
```diff
@@ -27,6 +27,7 @@ export const useLoadConnectors = (props: UseLoadConnectorsProps) => {
   const { data, isLoading, isFetching, isInitialLoading } = useQuery({
     queryKey: ['useLoadConnectors', includeSystemActions],
     queryFn,
+    cacheTime: 0,
```
What is the purpose of changing the cache time (garbage collection) to zero? Will this change affect other parts of the app?

cnasikas replied:
@JiaweiWu Any update on this?

JiaweiWu replied:
should be fine, nothing else is using this

cnasikas replied:
Yes, but why do we need to set it to `0`? What does it do and what issue does it solve?

cnasikas replied:
Ok, I spoke with @umbopepato, and help me understand. I made the hooks parameterized because they can be used in the future by other parts of the application and we may not want the caching to be set to zero.

cnasikas commented:
I found another bug. If I copy-paste the create URL in the browser, for example, `https://localhost:5602/app/management/insightsAndAlerting/triggersActions/rules/create/.index-threshold`, I get an error.

<img width="1284" alt="Screenshot 2024-10-08 at 9 55 30 AM" src="https://github.com/user-attachments/assets/bb8250e7-4743-4a62-b88f-b5aa681229c3">
<img width="517" alt="Screenshot 2024-10-08 at 9 55 21 AM" src="https://github.com/user-attachments/assets/1a583f1b-aa1d-4709-b3d0-764c65176d19">

guskovaue commented:
One more:
Cannot delete first number in `For the last` section using keyboard:

https://github.com/user-attachments/assets/85ac8ea5-e304-4bde-b0cd-67e037b1a884

guskovaue commented:
Also a little bit weird behaviour. 
Cannot remove 0 in input. And can have negative number as CPU %.

https://github.com/user-attachments/assets/7be11945-2ae6-47d4-9110-842aa0894d67

js-jankisalvi reviewed (commented):
I tested the `Log threshold`, `Anomaly detection jobs health`, `Elasticsearch version mismatch `and `Kibana version mismatch rule types` and the `JIRA`, `IBM resilient` and `The Hive` connector types. 
Everything is working as expected overall. New rule form is so smooth and easy to use 😍 🎉 

Found few small issues:
1. `Log threshold rule type` params validation works fine on `create` rule where as doesn't throw error on `edit` rule

https://github.com/user-attachments/assets/63cd64ae-2dee-4ebf-a806-69ba5e2f2444

2. Actions alerts filter throws error once toggled on and then toggled off. Same issue for `If alerts matches query` and `if alert is generated during timeframe`

https://github.com/user-attachments/assets/b64ec729-6506-422a-9101-b4aa16f7670b

3. Anomaly detection job health rule throws correct error when licence is not `Platinum`, however error says that the rule is disabled but you can see it is enabled.

<img width="676" alt="Screenshot 2024-10-07 at 14 48 07" src="https://github.com/user-attachments/assets/99075d9c-2b13-44a9-9cdf-b5b9f345b8fd">

4. This one might not be related to your PR but an issue with The Hive. The Hive connector behaves as it is (without any error) when licence is `Basic`, even though required licence for it is `Platinum`

https://github.com/user-attachments/assets/eb1f10d5-b296-4af0-96b9-d182c7570a09

jcger commented:
After creating a SLO I'm not able to create the rule. I also see no error in the form whatsoever

Steps:
- Go to SLO in o11y > Create SLO
- Click on "Select a Data View" > create new data view > Index pattern `.alerts-stack.alerts-default`, timefield: cases.updatedAt
- Click on good query > _id : * 
- set name, desc. and create
- Go to stack management, create SLO rule
- Select created SLO

The create button stays disabled whereas it works when using the flyout

jcger commented:
## Missing monitoring data

There are two options hidden that in the flyout show up the first time "save" is clicked. In the rules page the create button is disabled and these hidden options are not shown

<details>
  <summary>Flyout recording showing how two options show when save is clicked</summary>


https://github.com/user-attachments/assets/350bc5e3-7543-4ba0-8511-21bf26643cb3
</details>

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/240533)
* Commit: 1457db9b1a02745ad6978f0bca8dcab696d2b138

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/d5e2afdeee9a48d2da763fc6e300723cb855b953) | [after](https://github.com/elastic/kibana/commit/1457db9b1a02745ad6978f0bca8dcab696d2b138) | diff |
| --- | --- | --- | --- |
| `alerting` | 237 | 238 | +1 |
| `cases` | 812 | 813 | +1 |
| `observability` | 1061 | 1062 | +1 |
| `securitySolution` | 5925 | 5926 | +1 |
| `triggersActionsUi` | 798 | 844 | +46 |
| total |  |  | +50 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/d5e2afdeee9a48d2da763fc6e300723cb855b953) | [after](https://github.com/elastic/kibana/commit/1457db9b1a02745ad6978f0bca8dcab696d2b138) | diff |
| --- | --- | --- | --- |
| `@kbn/rule-data-utils` | 126 | 132 | +6 |
| `triggersActionsUi` | 566 | 567 | +1 |
| total |  |  | +7 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/d5e2afdeee9a48d2da763fc6e300723cb855b953) | [after](https://github.com/elastic/kibana/commit/1457db9b1a02745ad6978f0bca8dcab696d2b138) | diff |
| --- | --- | --- | --- |
| `triggersActionsUi` | 1.6MB | 1.7MB | +63.0KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/d5e2afdeee9a48d2da763fc6e300723cb855b953) | [after](https://github.com/elastic/kibana/commit/1457db9b1a02745ad6978f0bca8dcab696d2b138) | diff |
| --- | --- | --- | --- |
| `triggersActionsUi` | 123.3KB | 127.4KB | +4.1KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/d5e2afdeee9a48d2da763fc6e300723cb855b953) | [after](https://github.com/elastic/kibana/commit/1457db9b1a02745ad6978f0bca8dcab696d2b138) | diff |
| --- | --- | --- | --- |
| `@kbn/rule-data-utils` | 129 | 135 | +6 |
| `triggersActionsUi` | 592 | 593 | +1 |
| total |  |  | +7 |

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/d5e2afdeee9a48d2da763fc6e300723cb855b953) | [after](https://github.com/elastic/kibana/commit/1457db9b1a02745ad6978f0bca8dcab696d2b138) | diff |
| --- | --- | --- | --- |
| `triggersActionsUi` | 64 | 68 | +4 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/d5e2afdeee9a48d2da763fc6e300723cb855b953) | [after](https://github.com/elastic/kibana/commit/1457db9b1a02745ad6978f0bca8dcab696d2b138) | diff |
| --- | --- | --- | --- |
| `triggersActionsUi` | 127 | 130 | +3 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/d5e2afdeee9a48d2da763fc6e300723cb855b953) | [after](https://github.com/elastic/kibana/commit/1457db9b1a02745ad6978f0bca8dcab696d2b138) | diff |
| --- | --- | --- | --- |
| `triggersActionsUi` | 133 | 136 | +3 |


</details>


<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/240533","number":240533,"commit":"1457db9b1a02745ad6978f0bca8dcab696d2b138"}],"number":240533}
buildkite-pr-comment-->

umbopepato commented:
Rules tested:
- `Tracking containment`
- `Transform health`
- `Inventory`

Connectors tested:
- `Index`
- `Server log`
- `Webhook`

The new rule form is great, and I couldn't find any issue that seems to be directly related to this PR 🎉 

I did find though:
- [an issue with unprivileged users](https://github.com/elastic/kibana/issues/195426)
- apart from [what Janki already reported with action filters](https://github.com/elastic/kibana/pull/194655#pullrequestreview-2354553052), it seems like if you add a filter using the KQL bar (+ button) without typing any textual query, the validation fails thinking that the query is empty:
    <img width="542" alt="Screenshot 2024-10-09 at 11 25 47" src="https://github.com/user-attachments/assets/5c69cb9b-a3e6-451b-a626-3100cc8a24f4">

cnasikas commented:
> it seems like if you add a filter using the KQL bar (+ button) without typing any textual query, the validation fails thinking that the query is empty:

I think this is the same as https://github.com/elastic/kibana/issues/192557.

JiaweiWu commented:
@elasticmachine merge upstream

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/240957)
* Commit: 0ae2e997d88f620651f7aeda4708ed53d2e2c690

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/240957#0192742c-0d03-4db6-b9cd-d309b7ef7fbb)

### History
* :green_heart: [Build #240533](https://buildkite.com/elastic/kibana-pull-request/builds/240533) succeeded 1457db9b1a02745ad6978f0bca8dcab696d2b138

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/240957","number":240957,"commit":"0ae2e997d88f620651f7aeda4708ed53d2e2c690"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/240533","number":240533,"commit":"1457db9b1a02745ad6978f0bca8dcab696d2b138"}],"number":240957}
buildkite-pr-comment-->

JiaweiWu commented:
In my last PR (https://github.com/elastic/kibana/pull/194655/commits/d53d2223d52345971c07b846bb913a70b7c5070e) I addressed the following: 

@cnasikas:
- `When editing an Anomaly detection rule type I get the following error`  [Fixed]
- `The default value for the "Correlation ID" is not being populated for ServiceNow ITSM. For SecOps it works as expected.` [Fixed]
- `Breadcrumb are missing when creating or editing a rule. In Cases we append the "Create" word.` [Fixed]
- `I found another bug. If I copy-paste the create or the edit URL in the browser, for example, /app/management/insightsAndAlerting/triggersActions/rules/create/.index-threshold or /app/management/insightsAndAlerting/triggersActions/rules/edit/15353364-5d4d-43b0-bf1c-9afe573502a6, I get an error.` [Fixed]

@guskovaue:
- `During testing ES query rule, I found small issue. Probably it's like this for all rule types.
Look video please.
Not sure it's a bug, but behave differently in 2 similar cases.` [This one is not a bug, since it's the old flyout exhibiting this issue]
- `One more: Cannot delete first number in For the last section using keyboard:` [This is a rule type expression component bug, unrelated to the new rule form]
- `Also a little bit weird behaviour. Cannot remove 0 in input. And can have negative number as CPU % and time range.` [Same with this]

@js-jankisalvi:
- `Log threshold rule type params validation works fine on create rule where as doesn't throw error on edit rule` [Fixed]
- `Actions alerts filter throws error once toggled on and then toggled off. Same issue for If alerts matches query and if alert is generated during timeframe` [Fixed]
- `Anomaly detection job health rule throws correct error when licence is not Platinum, however error says that the rule is disabled but you can see it is enabled.` [This one seems intentional, not a bug I think]
- `This one might not be related to your PR but an issue with The Hive. The Hive connector behaves as it is (without any error) when licence is Basic, even though required licence for it is Platinum` [I was unable to reproduce this, but doesn't seem related to the rule form]

@jcger:
- `SLO burn rate bug` [Fixed]
- `Missing monitoring data` [This one is interesting, the only reason why it works right now is by accident, we should think about re-writing the origin rule type expression]

@umbopepato:
- `https://github.com/elastic/kibana/issues/195426` [This one seems unrelated to the rule form]
- `it seems like if you add a filter using the KQL bar (+ button) without typing any textual query, the validation fails thinking that the query is empty:` [Fixed]

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/241428)
* Commit: d53d2223d52345971c07b846bb913a70b7c5070e

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/241428#01927a6d-845b-410f-a1fa-63548150a3c1)

### History
* :broken_heart: [Build #240957](https://buildkite.com/elastic/kibana-pull-request/builds/240957) failed 0ae2e997d88f620651f7aeda4708ed53d2e2c690
* :green_heart: [Build #240533](https://buildkite.com/elastic/kibana-pull-request/builds/240533) succeeded 1457db9b1a02745ad6978f0bca8dcab696d2b138

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241428","number":241428,"commit":"d53d2223d52345971c07b846bb913a70b7c5070e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/240957","number":240957,"commit":"0ae2e997d88f620651f7aeda4708ed53d2e2c690"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/240533","number":240533,"commit":"1457db9b1a02745ad6978f0bca8dcab696d2b138"}],"number":241428}
buildkite-pr-comment-->

js-jankisalvi commented:
> @js-jankisalvi:
> *  Found an issue for alert delay, I think it's an issue for all rule types. (tested it in log threshold and anomaly detection job health rule types).UI allows to remove value from alert delay but when you save the rule, previous value is saved. ✅ 
> * `Log threshold rule type params validation works fine on create rule where as doesn't throw error on edit rule` [Fixed] ❌ 
> * `Actions alerts filter throws error once toggled on and then toggled off. Same issue for If alerts matches query and if alert is generated during timeframe` [Fixed] ✅ 
> * `Anomaly detection job health rule throws correct error when licence is not Platinum, however error says that the rule is disabled but you can see it is enabled.` [This one seems intentional, not a bug I think] 👍 
> * `This one might not be related to your PR but an issue with The Hive. The Hive connector behaves as it is (without any error) when licence is Basic, even though required licence for it is Platinum` [I was unable to reproduce this, but doesn't seem related to the rule form] 👍 

Log threshold rule type validation still didn't throw any error for edit rule and allowed to save. However I saw similar issue in `old rule flyout` as well.

<img width="705" alt="Screenshot 2024-10-11 at 10 20 19" src="https://github.com/user-attachments/assets/cdf567be-1fcf-4aac-8e10-f59ac6393071">

<img width="707" alt="Screenshot 2024-10-11 at 11 26 07" src="https://github.com/user-attachments/assets/97212963-31e3-478b-a26c-74680fa19a0a">

adcoelho commented:
I tested `Thread pool write rejections`, `CCR read exceptions`, and `Shard size`. For the most part, they are ok but I ran into an annoying scenario.

1. I created one of each of these rule types in `main`
2. I got your PR and then tried to edit these preexisting rules
3. The Save button was disabled but the form seemed fine. 

This was frustrating because there was no error in the form.
<img width="1191" alt="Screenshot 2024-10-11 at 14 28 51" src="https://github.com/user-attachments/assets/de536be9-ae3a-4f74-8ed5-50f5b50444dd">


**Except well, there was but it is hidden in the `Advanced options`.**


<img width="1076" alt="Screenshot 2024-10-11 at 14 29 01" src="https://github.com/user-attachments/assets/d5bba307-0969-4a7a-b148-e43fa795ec0a">

cnasikas commented:
I can verify that all the bugs are fixed. I found the following ones:

- If I do not have permission to the "Rule Settings" the flapping section is being shown but the right part is empty. I would say to hide the whole section entirely. Wdyt?

<img width="1149" alt="Screenshot 2024-10-11 at 4 27 00 PM" src="https://github.com/user-attachments/assets/238b9a6b-9820-4932-84ba-3e34e33e5a52">

- If I do not have permission to "Actions" the modal appears to inform me about a rule with no actions. If I create and then edit I do not see it again. I think we should not show it as I cannot do anything as a user. Wdyt?

<img width="1237" alt="Screenshot 2024-10-11 at 4 34 19 PM" src="https://github.com/user-attachments/assets/0e61de75-9c47-42fe-8d15-5ee291e02d1c">

- Some connectors are not available on the basic license. If there is a rule with connectors available on a greater license than basic and you are on basic you can see them but you cannot save the rule. In `main` the following is being shown:

<img width="597" alt="Screenshot 2024-10-11 at 5 04 20 PM" src="https://github.com/user-attachments/assets/cec2af40-bd54-4d1e-b49e-568f60c487bf">
<img width="616" alt="Screenshot 2024-10-11 at 5 04 35 PM" src="https://github.com/user-attachments/assets/440e3bc8-511b-4748-82a2-ba7f36aef41f">

and I can save the rule if I want. Should we do something similar on the new rule form?

- On `main` if I press cancel or exit the flyout I get the following callout 

<img width="532" alt="Screenshot 2024-10-11 at 5 05 15 PM" src="https://github.com/user-attachments/assets/cdb0a92d-1a44-48ff-87a3-481426fcd8d3">

Should we do something similar on the new rule form (cancel or return button)?

- On basic license, if I hover on unsupported connectors a tooltip is being shown explaining the reasoning. This is not happening for rule types. Should we do the same? Probably unrelated to this PR.

<img width="601" alt="Screenshot 2024-10-11 at 5 13 23 PM" src="https://github.com/user-attachments/assets/3b4876e8-b6f7-4661-a359-75dea4b25a23">
<img width="832" alt="Screenshot 2024-10-11 at 5 13 41 PM" src="https://github.com/user-attachments/assets/6f6d8a42-7847-4198-8ed2-b672a7c13032">

- On basic license, I can edit a rule that is not available on the basic. This leads to a bunch of errors in edit mode. The same is happening on `main`. Should we prevent accessing this kind of rule types (for another PR)?

yctercero reviewed (approved):
Codeowner changes `packages/kbn-rule-data-utils/src/routes/stack_rule_paths.ts` LGTM.

JiaweiWu commented:
In my last commit (https://github.com/elastic/kibana/pull/194655/commits/cb8f068574bf2fef8c2677ee016e5b72b5bd5848 I addressed the following:

@jcger:

- `There are two options hidden that in the flyout show up the first time "save" is clicked. In the rules page the create button is disabled and these hidden options are not shown` [Fixed]

@js-jankisalvi:

- `Log threshold rule type validation still didn't throw any error for edit rule and allowed to save. However I saw similar issue in old rule flyout as well.` [This seems to be the existing behaviour, so I won't fix it here, but please create a ticket to fix later]

@adcoelho: 

- `This was frustrating because there was no error in the form.` [Fixed, added danger status to the step indicating there are errors]
<img width="987" alt="Screenshot 2024-10-11 at 10 05 20 PM" src="https://github.com/user-attachments/assets/8b617fb1-b76a-4a38-9f24-d23f1297d615">

@cnasikas:

- `If I do not have permission to the "Rule Settings" the flapping section is being shown but the right part is empty. I would say to hide the whole section entirely. Wdyt?` [Fixed]
- `If I do not have permission to "Actions" the modal appears to inform me about a rule with no actions. If I create and then edit I do not see it again. I think we should not show it as I cannot do anything as a user. Wdyt?` [Fixed]
- `Some connectors are not available on the basic license. If there is a rule with connectors available on a greater license than basic and you are on basic you can see them but you cannot save the rule. In main the following is being shown:` [Fixed]
<img width="864" alt="Screenshot 2024-10-11 at 9 35 01 PM" src="https://github.com/user-attachments/assets/e23162ec-9b75-4a70-8cce-52b843f28c0b">
<img width="870" alt="Screenshot 2024-10-11 at 9 46 50 PM" src="https://github.com/user-attachments/assets/01e18702-52dd-4164-8287-5059fb5881b0">

- `On main if I press cancel or exit the flyout I get the following callout` [Fixed]
- `On basic license, if I hover on unsupported connectors a tooltip is being shown explaining the reasoning. This is not happening for rule types. Should we do the same? Probably unrelated to this PR.` [Looked like an improvement for the rule type modal, I won't address it in this PR]
- `On basic license, I can edit a rule that is not available on the basic. This leads to a bunch of errors in edit mode. The same is happening on main. Should we prevent accessing this kind of rule types (for another PR)?` [Happening in main, therefore won't address it in this PR]

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/241834)
* Commit: cb8f068574bf2fef8c2677ee016e5b72b5bd5848

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/241834#01927f20-a074-4e4f-ac73-4f0abafa9c53)

### History
* :broken_heart: [Build #241428](https://buildkite.com/elastic/kibana-pull-request/builds/241428) failed d53d2223d52345971c07b846bb913a70b7c5070e
* :broken_heart: [Build #240957](https://buildkite.com/elastic/kibana-pull-request/builds/240957) failed 0ae2e997d88f620651f7aeda4708ed53d2e2c690
* :green_heart: [Build #240533](https://buildkite.com/elastic/kibana-pull-request/builds/240533) succeeded 1457db9b1a02745ad6978f0bca8dcab696d2b138

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241834","number":241834,"commit":"cb8f068574bf2fef8c2677ee016e5b72b5bd5848"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241428","number":241428,"commit":"d53d2223d52345971c07b846bb913a70b7c5070e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/240957","number":240957,"commit":"0ae2e997d88f620651f7aeda4708ed53d2e2c690"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/240533","number":240533,"commit":"1457db9b1a02745ad6978f0bca8dcab696d2b138"}],"number":241834}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/241878)
* Commit: 70c8baacecad457760e5004cbcb5622e4cb9f8f7

### Failed CI Steps
* [x-pack/test_serverless/functional/test_suites/security/common_configs/config.group6.ts](https://buildkite.com/elastic/kibana-pull-request/builds/241878#0192852b-7647-42d9-86ef-327a5813e0c4)
* [FTR Configs #35](https://buildkite.com/elastic/kibana-pull-request/builds/241878#0192852b-76a5-4a13-996b-70e000dd4204)
* [FTR Configs #51](https://buildkite.com/elastic/kibana-pull-request/builds/241878#0192852b-76c8-433b-bbb5-270247f26909)
* [FTR Configs #75](https://buildkite.com/elastic/kibana-pull-request/builds/241878#0192852b-7723-4b55-b68e-a38f6c71b1a0)
* [FTR Configs #81](https://buildkite.com/elastic/kibana-pull-request/builds/241878#0192852b-7735-4386-a903-6acc072794c3)
* [FTR Configs #81](https://buildkite.com/elastic/kibana-pull-request/builds/241878#01928542-3628-4d30-8caf-e0e40312b418)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/241878#0192852b-7735-4386-a903-6acc072794c3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/241878/jobs/0192852b-7735-4386-a903-6acc072794c3/artifacts/01928542-24fc-443c-9456-588e9776c83f) FTR Configs #<span></span>81 / Actions and Triggers app create alert should delete the right action when the same action has been added twice
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/241878#01928542-3628-4d30-8caf-e0e40312b418) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/241878/jobs/01928542-3628-4d30-8caf-e0e40312b418/artifacts/0192854b-ee2c-4a61-9d0e-dec7cd396946) FTR Configs #<span></span>81 / Actions and Triggers app create alert should delete the right action when the same action has been added twice
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/241878#0192852b-7647-42d9-86ef-327a5813e0c4) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/241878/jobs/0192852b-7647-42d9-86ef-327a5813e0c4/artifacts/01928545-a4fb-456c-ab15-87a57749a07f) x-pack/test_serverless/functional/test_suites/security/common_configs/config.group6.ts / Discover alerting Search source Alert should display prev rule state after params update on clicking prev generated link
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/241878#0192852b-76a5-4a13-996b-70e000dd4204) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/241878/jobs/0192852b-76a5-4a13-996b-70e000dd4204/artifacts/01928548-7f24-4086-8661-74e56106d3dc) FTR Configs #<span></span>35 / Discover alerting Search source Alert should display prev rule state after params update on clicking prev generated link
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/241878#0192852b-76c8-433b-bbb5-270247f26909) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/241878/jobs/0192852b-76c8-433b-bbb5-270247f26909/artifacts/01928546-6eed-440e-981d-f5b202e6715e) FTR Configs #<span></span>51 / Discover alerting Search source Alert should display prev rule state after params update on clicking prev generated link
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/241878#0192852b-7723-4b55-b68e-a38f6c71b1a0) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/241878/jobs/0192852b-7723-4b55-b68e-a38f6c71b1a0/artifacts/01928548-0689-4bb1-813b-525740f6d7c3) FTR Configs #<span></span>75 / Discover alerting Search source Alert should display prev rule state after params update on clicking prev generated link

### History
* :broken_heart: [Build #241834](https://buildkite.com/elastic/kibana-pull-request/builds/241834) failed cb8f068574bf2fef8c2677ee016e5b72b5bd5848
* :broken_heart: [Build #241428](https://buildkite.com/elastic/kibana-pull-request/builds/241428) failed d53d2223d52345971c07b846bb913a70b7c5070e
* :broken_heart: [Build #240957](https://buildkite.com/elastic/kibana-pull-request/builds/240957) failed 0ae2e997d88f620651f7aeda4708ed53d2e2c690
* :green_heart: [Build #240533](https://buildkite.com/elastic/kibana-pull-request/builds/240533) succeeded 1457db9b1a02745ad6978f0bca8dcab696d2b138

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241878","number":241878,"commit":"70c8baacecad457760e5004cbcb5622e4cb9f8f7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241834","number":241834,"commit":"cb8f068574bf2fef8c2677ee016e5b72b5bd5848"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241428","number":241428,"commit":"d53d2223d52345971c07b846bb913a70b7c5070e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/240957","number":240957,"commit":"0ae2e997d88f620651f7aeda4708ed53d2e2c690"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/240533","number":240533,"commit":"1457db9b1a02745ad6978f0bca8dcab696d2b138"}],"number":241878}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/241908)
* Commit: b570c708d7dbbd9930e9670940d61e26b5051b01
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-194655/b570c708d7dbbd9930e9670940d61e26b5051b01)

### Failed CI Steps
* [FTR Configs #91](https://buildkite.com/elastic/kibana-pull-request/builds/241908#01928815-5e61-4d44-924e-aa3dba722a9b)
* [FTR Configs #91](https://buildkite.com/elastic/kibana-pull-request/builds/241908#0192883d-dbf5-478a-a11b-2689abb93619)

### History
* :broken_heart: [Build #241878](https://buildkite.com/elastic/kibana-pull-request/builds/241878) failed 70c8baacecad457760e5004cbcb5622e4cb9f8f7
* :broken_heart: [Build #241834](https://buildkite.com/elastic/kibana-pull-request/builds/241834) failed cb8f068574bf2fef8c2677ee016e5b72b5bd5848
* :broken_heart: [Build #241428](https://buildkite.com/elastic/kibana-pull-request/builds/241428) failed d53d2223d52345971c07b846bb913a70b7c5070e
* :broken_heart: [Build #240957](https://buildkite.com/elastic/kibana-pull-request/builds/240957) failed 0ae2e997d88f620651f7aeda4708ed53d2e2c690
* :green_heart: [Build #240533](https://buildkite.com/elastic/kibana-pull-request/builds/240533) succeeded 1457db9b1a02745ad6978f0bca8dcab696d2b138

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241908","number":241908,"commit":"b570c708d7dbbd9930e9670940d61e26b5051b01"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241878","number":241878,"commit":"70c8baacecad457760e5004cbcb5622e4cb9f8f7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241834","number":241834,"commit":"cb8f068574bf2fef8c2677ee016e5b72b5bd5848"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241428","number":241428,"commit":"d53d2223d52345971c07b846bb913a70b7c5070e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/240957","number":240957,"commit":"0ae2e997d88f620651f7aeda4708ed53d2e2c690"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/240533","number":240533,"commit":"1457db9b1a02745ad6978f0bca8dcab696d2b138"}],"number":241908}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/241914)
* Commit: 15a54275022e4b509e365f7d99705e736b17488e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-194655/15a54275022e4b509e365f7d99705e736b17488e)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/288d41d61ec2389b2e8856da75fd0f3107f9c484) | [after](https://github.com/elastic/kibana/commit/15a54275022e4b509e365f7d99705e736b17488e) | diff |
| --- | --- | --- | --- |
| `triggersActionsUi` | 805 | 850 | +45 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/288d41d61ec2389b2e8856da75fd0f3107f9c484) | [after](https://github.com/elastic/kibana/commit/15a54275022e4b509e365f7d99705e736b17488e) | diff |
| --- | --- | --- | --- |
| `@kbn/alerts-ui-shared` | 305 | 302 | -3 |
| `@kbn/rule-data-utils` | 126 | 132 | +6 |
| `triggersActionsUi` | 566 | 567 | +1 |
| total |  |  | +4 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/288d41d61ec2389b2e8856da75fd0f3107f9c484) | [after](https://github.com/elastic/kibana/commit/15a54275022e4b509e365f7d99705e736b17488e) | diff |
| --- | --- | --- | --- |
| `triggersActionsUi` | 1.6MB | 1.7MB | +81.7KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/288d41d61ec2389b2e8856da75fd0f3107f9c484) | [after](https://github.com/elastic/kibana/commit/15a54275022e4b509e365f7d99705e736b17488e) | diff |
| --- | --- | --- | --- |
| `triggersActionsUi` | 123.3KB | 127.3KB | +4.0KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/288d41d61ec2389b2e8856da75fd0f3107f9c484) | [after](https://github.com/elastic/kibana/commit/15a54275022e4b509e365f7d99705e736b17488e) | diff |
| --- | --- | --- | --- |
| `@kbn/alerts-ui-shared` | 321 | 318 | -3 |
| `@kbn/rule-data-utils` | 129 | 135 | +6 |
| `triggersActionsUi` | 592 | 593 | +1 |
| total |  |  | +4 |

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/288d41d61ec2389b2e8856da75fd0f3107f9c484) | [after](https://github.com/elastic/kibana/commit/15a54275022e4b509e365f7d99705e736b17488e) | diff |
| --- | --- | --- | --- |
| `triggersActionsUi` | 64 | 66 | +2 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/288d41d61ec2389b2e8856da75fd0f3107f9c484) | [after](https://github.com/elastic/kibana/commit/15a54275022e4b509e365f7d99705e736b17488e) | diff |
| --- | --- | --- | --- |
| `@kbn/alerts-ui-shared` | 7 | 8 | +1 |
| `triggersActionsUi` | 127 | 130 | +3 |
| total |  |  | +4 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/288d41d61ec2389b2e8856da75fd0f3107f9c484) | [after](https://github.com/elastic/kibana/commit/15a54275022e4b509e365f7d99705e736b17488e) | diff |
| --- | --- | --- | --- |
| `@kbn/alerts-ui-shared` | 8 | 9 | +1 |
| `triggersActionsUi` | 133 | 136 | +3 |
| total |  |  | +4 |


</details>


### History
* :broken_heart: [Build #241908](https://buildkite.com/elastic/kibana-pull-request/builds/241908) failed b570c708d7dbbd9930e9670940d61e26b5051b01
* :broken_heart: [Build #241878](https://buildkite.com/elastic/kibana-pull-request/builds/241878) failed 70c8baacecad457760e5004cbcb5622e4cb9f8f7
* :broken_heart: [Build #241834](https://buildkite.com/elastic/kibana-pull-request/builds/241834) failed cb8f068574bf2fef8c2677ee016e5b72b5bd5848
* :broken_heart: [Build #241428](https://buildkite.com/elastic/kibana-pull-request/builds/241428) failed d53d2223d52345971c07b846bb913a70b7c5070e
* :broken_heart: [Build #240957](https://buildkite.com/elastic/kibana-pull-request/builds/240957) failed 0ae2e997d88f620651f7aeda4708ed53d2e2c690

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241914","number":241914,"commit":"15a54275022e4b509e365f7d99705e736b17488e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241908","number":241908,"commit":"b570c708d7dbbd9930e9670940d61e26b5051b01"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241878","number":241878,"commit":"70c8baacecad457760e5004cbcb5622e4cb9f8f7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241834","number":241834,"commit":"cb8f068574bf2fef8c2677ee016e5b72b5bd5848"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241428","number":241428,"commit":"d53d2223d52345971c07b846bb913a70b7c5070e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/240957","number":240957,"commit":"0ae2e997d88f620651f7aeda4708ed53d2e2c690"}],"number":241914}
buildkite-pr-comment-->

js-jankisalvi commented:
> @js-jankisalvi:
> 
> * `Log threshold rule type validation still didn't throw any error for edit rule and allowed to save. However I saw similar issue in old rule flyout as well.` [This seems to be the existing behaviour, so I won't fix it here, but please create a ticket to fix later]

[Created issue](https://github.com/elastic/kibana/issues/196065)

jcger reviewed (approved):
LGTM 🍠

maryam-saeidi reviewed (approved):
Changes LGTM!

I tested the issue that I mentioned after this [commit](https://github.com/elastic/kibana/pull/194655/commits/1457db9b1a02745ad6978f0bca8dcab696d2b138), and I was able to see the page.

pheyos reviewed (approved):
`x-pack/test_serverless/functional/config.base.ts` changes LGTM

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242270)
* Commit: 4fdf3a79abef69c2e39cebcfff2d6974df839cea
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-194655/4fdf3a79abef69c2e39cebcfff2d6974df839cea)

### Failed CI Steps
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/242270#01928be6-c7c6-4b54-a591-d17b34fc3a7b)
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/242270#01928c19-1d3a-4663-b924-f04e3b2b3349)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/242270#01928be6-c7c6-4b54-a591-d17b34fc3a7b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/242270/jobs/01928be6-c7c6-4b54-a591-d17b34fc3a7b/artifacts/01928c18-dd85-47a6-b996-31f089121911) Jest Tests #<span></span>1 / useLoadDependencies should call resolve rule with the correct params
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/242270#01928c19-1d3a-4663-b924-f04e3b2b3349) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/242270/jobs/01928c19-1d3a-4663-b924-f04e3b2b3349/artifacts/01928c20-bf05-4766-843c-26db99ced862) Jest Tests #<span></span>1 / useLoadDependencies should call resolve rule with the correct params

### History
* :green_heart: [Build #241914](https://buildkite.com/elastic/kibana-pull-request/builds/241914) succeeded 15a54275022e4b509e365f7d99705e736b17488e
* :broken_heart: [Build #241908](https://buildkite.com/elastic/kibana-pull-request/builds/241908) failed b570c708d7dbbd9930e9670940d61e26b5051b01
* :broken_heart: [Build #241878](https://buildkite.com/elastic/kibana-pull-request/builds/241878) failed 70c8baacecad457760e5004cbcb5622e4cb9f8f7
* :broken_heart: [Build #241834](https://buildkite.com/elastic/kibana-pull-request/builds/241834) failed cb8f068574bf2fef8c2677ee016e5b72b5bd5848
* :broken_heart: [Build #241428](https://buildkite.com/elastic/kibana-pull-request/builds/241428) failed d53d2223d52345971c07b846bb913a70b7c5070e
* :broken_heart: [Build #240957](https://buildkite.com/elastic/kibana-pull-request/builds/240957) failed 0ae2e997d88f620651f7aeda4708ed53d2e2c690

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242270","number":242270,"commit":"4fdf3a79abef69c2e39cebcfff2d6974df839cea"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241914","number":241914,"commit":"15a54275022e4b509e365f7d99705e736b17488e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241908","number":241908,"commit":"b570c708d7dbbd9930e9670940d61e26b5051b01"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241878","number":241878,"commit":"70c8baacecad457760e5004cbcb5622e4cb9f8f7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241834","number":241834,"commit":"cb8f068574bf2fef8c2677ee016e5b72b5bd5848"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241428","number":241428,"commit":"d53d2223d52345971c07b846bb913a70b7c5070e"}],"number":242270}
buildkite-pr-comment-->

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/242427)
* Commit: a7f76f9a3b4a616ba0a6e137ff6523a1946c297a
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-194655/a7f76f9a3b4a616ba0a6e137ff6523a1946c297a)

### Failed CI Steps
* [FTR Configs #74](https://buildkite.com/elastic/kibana-pull-request/builds/242427#01928f27-441d-41c4-8431-b5cf2763c008)
* [FTR Configs #103](https://buildkite.com/elastic/kibana-pull-request/builds/242427#01928f27-4458-4d42-aa47-687bf45470ec)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/242427#01928f27-4458-4d42-aa47-687bf45470ec) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/242427/jobs/01928f27-4458-4d42-aa47-687bf45470ec/artifacts/01928f4c-bf3d-488d-93de-f9534af76e6f) FTR Configs #<span></span>103 / Serverless security API cloud_security_posture Intercept the usage API request sent by the metering background task manager Should intercept usage API request for CSPM

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/563910b672b6dbe4f9e7931e36ec41e674fe8eb3) | [after](https://github.com/elastic/kibana/commit/a7f76f9a3b4a616ba0a6e137ff6523a1946c297a) | diff |
| --- | --- | --- | --- |
| `triggersActionsUi` | 805 | 850 | +45 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/563910b672b6dbe4f9e7931e36ec41e674fe8eb3) | [after](https://github.com/elastic/kibana/commit/a7f76f9a3b4a616ba0a6e137ff6523a1946c297a) | diff |
| --- | --- | --- | --- |
| `@kbn/alerts-ui-shared` | 305 | 304 | -1 |
| `@kbn/rule-data-utils` | 127 | 133 | +6 |
| `triggersActionsUi` | 567 | 568 | +1 |
| total |  |  | +6 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/563910b672b6dbe4f9e7931e36ec41e674fe8eb3) | [after](https://github.com/elastic/kibana/commit/a7f76f9a3b4a616ba0a6e137ff6523a1946c297a) | diff |
| --- | --- | --- | --- |
| `triggersActionsUi` | 1.6MB | 1.7MB | +81.9KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/563910b672b6dbe4f9e7931e36ec41e674fe8eb3) | [after](https://github.com/elastic/kibana/commit/a7f76f9a3b4a616ba0a6e137ff6523a1946c297a) | diff |
| --- | --- | --- | --- |
| `triggersActionsUi` | 123.3KB | 127.4KB | +4.1KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/563910b672b6dbe4f9e7931e36ec41e674fe8eb3) | [after](https://github.com/elastic/kibana/commit/a7f76f9a3b4a616ba0a6e137ff6523a1946c297a) | diff |
| --- | --- | --- | --- |
| `@kbn/alerts-ui-shared` | 321 | 320 | -1 |
| `@kbn/rule-data-utils` | 130 | 136 | +6 |
| `triggersActionsUi` | 593 | 594 | +1 |
| total |  |  | +6 |

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/563910b672b6dbe4f9e7931e36ec41e674fe8eb3) | [after](https://github.com/elastic/kibana/commit/a7f76f9a3b4a616ba0a6e137ff6523a1946c297a) | diff |
| --- | --- | --- | --- |
| `triggersActionsUi` | 64 | 66 | +2 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/563910b672b6dbe4f9e7931e36ec41e674fe8eb3) | [after](https://github.com/elastic/kibana/commit/a7f76f9a3b4a616ba0a6e137ff6523a1946c297a) | diff |
| --- | --- | --- | --- |
| `@kbn/alerts-ui-shared` | 7 | 8 | +1 |
| `triggersActionsUi` | 127 | 130 | +3 |
| total |  |  | +4 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/563910b672b6dbe4f9e7931e36ec41e674fe8eb3) | [after](https://github.com/elastic/kibana/commit/a7f76f9a3b4a616ba0a6e137ff6523a1946c297a) | diff |
| --- | --- | --- | --- |
| `@kbn/alerts-ui-shared` | 8 | 9 | +1 |
| `triggersActionsUi` | 133 | 136 | +3 |
| total |  |  | +4 |


</details>


### History
* :broken_heart: [Build #242270](https://buildkite.com/elastic/kibana-pull-request/builds/242270) failed 4fdf3a79abef69c2e39cebcfff2d6974df839cea
* :green_heart: [Build #241914](https://buildkite.com/elastic/kibana-pull-request/builds/241914) succeeded 15a54275022e4b509e365f7d99705e736b17488e
* :broken_heart: [Build #241908](https://buildkite.com/elastic/kibana-pull-request/builds/241908) failed b570c708d7dbbd9930e9670940d61e26b5051b01
* :broken_heart: [Build #241878](https://buildkite.com/elastic/kibana-pull-request/builds/241878) failed 70c8baacecad457760e5004cbcb5622e4cb9f8f7
* :broken_heart: [Build #241834](https://buildkite.com/elastic/kibana-pull-request/builds/241834) failed cb8f068574bf2fef8c2677ee016e5b72b5bd5848

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242427","number":242427,"commit":"a7f76f9a3b4a616ba0a6e137ff6523a1946c297a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/242270","number":242270,"commit":"4fdf3a79abef69c2e39cebcfff2d6974df839cea"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241914","number":241914,"commit":"15a54275022e4b509e365f7d99705e736b17488e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241908","number":241908,"commit":"b570c708d7dbbd9930e9670940d61e26b5051b01"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241878","number":241878,"commit":"70c8baacecad457760e5004cbcb5622e4cb9f8f7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241834","number":241834,"commit":"cb8f068574bf2fef8c2677ee016e5b72b5bd5848"}],"number":242427}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/11344036469

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.x|**Backport failed because of merge conflicts**<br><br>You might need to backport the following PRs to 8.x:<br> - [[FTR] support custom native roles in serverless tests (#194677)](https://github.com/elastic/kibana/pull/194677)|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 194655
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->

cnasikas commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/196290">](https://github.com/elastic/kibana/pull/196290)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.8 -->
